DownloadTileCacheSample/swift/DownloadTileCache/ViewController.swift

```diff
@@ -0,0 +1,252 @@
+//
+// Copyright 2014 ESRI
+//
+// All rights reserved under the copyright laws of the United States
+// and applicable international laws, treaties, and conventions.
+//
+// You may freely redistribute and use this sample code, with or
+// without modification, provided you include the original copyright
+// notice and use restrictions.
+//
+// See the use restrictions at http://help.arcgis.com/en/sdk/10.0/usageRestrictions.htm
+//
+
+import UIKit
+import ArcGIS
+
+class ViewController: UIViewController, AGSLayerDelegate {
+    
+    @IBOutlet weak var mapView:AGSMapView!
+    @IBOutlet weak var downloadPanel:UIView!
+    @IBOutlet weak var scaleLabel:UILabel!
+    @IBOutlet  var estimateLabel:UILabel!
+    @IBOutlet weak var lodLabel:UILabel!
+    @IBOutlet weak var estimateButton:UIButton!
+    @IBOutlet weak var downloadButton:UIButton!
+    @IBOutlet weak var levelStepper:UIStepper!
+    @IBOutlet weak var timerLabel:UILabel!
+    
+    var tileCacheTask:AGSExportTileCacheTask!
+    var tiledLayer:AGSTiledMapServiceLayer!
+    
+    // in iOS7 this gets called and hides the status bar so the view does not go under the top iPhone status bar
+    override func prefersStatusBarHidden() -> Bool {
+        return true
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        //You can change this to any other service on tiledbasemaps.arcgis.com if you have an ArcGIS for Organizations subscription
+        let tileServiceURL = "http://sampleserver6.arcgisonline.com/arcgis/rest/services/World_Street_Map/MapServer"
+        
+        //Add basemap layer to the map
+        //Set delegate to be notified of success or failure while loading
+        let tiledUrl = NSURL(string: tileServiceURL)
+        self.tiledLayer = AGSTiledMapServiceLayer(URL: tiledUrl)
+        self.tiledLayer.delegate  = self
+        self.mapView.addMapLayer(self.tiledLayer, withName:"World Street Map")
+        
+        
+        // Init the tile cache task
+        if self.tileCacheTask == nil {
+            self.tileCacheTask = AGSExportTileCacheTask(URL: tiledUrl)
+        }
+        
+        self.scaleLabel.numberOfLines = 0
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    //MARK: - AGSLayer delegate
+    
+    func layer(layer: AGSLayer!, didFailToLoadWithError error: NSError!) {
+        //Alert user of error
+        UIAlertView(title: "Error", message: error.localizedDescription, delegate: nil, cancelButtonTitle: nil).show()
+    }
+    
+    func layerDidLoad(layer: AGSLayer!) {
+        if layer == self.tiledLayer {
+            //Initialize UIStepper based on number of scale levels in the tiled layer
+            self.levelStepper.value = 0
+            self.levelStepper.minimumValue = 0
+            self.levelStepper.maximumValue = Double(self.tiledLayer.tileInfo.lods.count)
```

Comment:
Updated the code!

From: Divesh Goyal <notifications@github.com<mailto:notifications@github.com>>
Reply-To: Esri/arcgis-runtime-samples-ios <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Wednesday, January 7, 2015 at 3:35 PM
To: Esri/arcgis-runtime-samples-ios <arcgis-runtime-samples-ios@noreply.github.com<mailto:arcgis-runtime-samples-ios@noreply.github.com>>
Cc: Gagandeep Singh <gagandeep_singh@esri.com<mailto:gagandeep_singh@esri.com>>
Subject: Re: [arcgis-runtime-samples-ios] added the swift version for DownloadTileCache sample and update objc fol... (#96)


In DownloadTileCacheSample/swift/DownloadTileCache/ViewController.swift<https://github.com/Esri/arcgis-runtime-samples-ios/pull/96#discussion-diff-22625673>:

> +        // Dispose of any resources that can be recreated.
> +    }
> +
> +    //MARK: - AGSLayer delegate
> +
> +    func layer(layer: AGSLayer!, didFailToLoadWithError error: NSError!) {
> +        //Alert user of error
> +        UIAlertView(title: "Error", message: error.localizedDescription, delegate: nil, cancelButtonTitle: nil).show()
> +    }
> +
> +    func layerDidLoad(layer: AGSLayer!) {
> +        if layer == self.tiledLayer {
> +            //Initialize UIStepper based on number of scale levels in the tiled layer
> +            self.levelStepper.value = 0
> +            self.levelStepper.minimumValue = 0
> +            self.levelStepper.maximumValue = Double(self.tiledLayer.tileInfo.lods.count)


This should be count-1
See #91<https://github.com/Esri/arcgis-runtime-samples-ios/pull/91>

—
Reply to this email directly or view it on GitHub<https://github.com/Esri/arcgis-runtime-samples-ios/pull/96/files#r22625673>.